### DISCUSSION TITLE
@types/consul is outdated

### DISCUSSION THREAD

**Discussion by containerman17** [DefinitelyTyped/DefinitelyTyped#53957] [Issues with a @types package]
Hi! 👋 
      
Firstly, thanks for your work on this project! 🙂

Today I used [patch-package](https://github.com/ds300/patch-package) to patch `@types/consul@0.23.34` for the project I'm working on.

This is not a full update but that helps with responses and recurse watch

Here is the diff that solved my problem:

```diff
diff --git a/node_modules/@types/consul/index.d.ts b/node_modules/@types/consul/index.d.ts
index 18355be..d5f3b48 100644
--- a/node_modules/@types/consul/index.d.ts
+++ b/node_modules/@types/consul/index.d.ts
@@ -719,6 +719,11 @@ declare namespace Consul {
         }
 
         interface DeleteOptions extends DelOptions { }
+
+        interface GetOneResponse {
+            Value: string,
+            ModifyIndex: string,
+        }
     }
 
     interface Kv {
@@ -956,6 +961,7 @@ declare namespace Consul {
         
         interface WatchOptions {
             key?: string;
+            recurse?: boolean;
         }
 
         interface Options {
```

<em>This issue body was [partially generated by patch-package](https://github.com/ds300/patch-package/issues/296).</em>

**Comment by UNIDY2002** [comment]
Good news is that `consul@2.0.0-next.1` now includes its own typing. While still not perfect, it is worth having a try.
